templates/static/js/example.js

```diff
@@ -38,32 +80,72 @@
     [].slice.call(examples).forEach(fetchExample);
   });
 
-  function fetchExample(exampleElement) {
-    var link = exampleElement.href;
+  /**
+   * Sends a `GET` request to `url` to request an example's contents.
+   * @param {String} url Address of the example
+   * @returns {Promise<String>} Response text
+   */
+  async function fetchExampleResponseText(url) {
     var request = new XMLHttpRequest();
 
-    request.onreadystatechange = function () {
-      if (request.status === 200 && request.readyState === 4) {
-        var html = request.responseText;
-        renderExample(exampleElement, html);
-        exampleElement.style.display = 'none';
-      }
-    };
+    const result = new Promise(function (resolve, reject) {
+      request.onreadystatechange = function () {
+        if (request.status === 200 && request.readyState === 4) {
+          resolve(request.responseText);
+        } else if (request.status > 0 && (request.status < 200 || request.status >= 300)) {
```

Comment:
This was written this way because [onreadystatechange](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/readyState) is fired 4 times per request. I simplified it to avoid that difficult-to-read `else if` condition by checking for incomplete requests at the start of the callback.